### PR TITLE
 comparing and printing local and registry version of CB CLI

### PR DESCRIPTION
**Pull request by shahraiz-cb**
Crowdbotics CLI

## Ticket

ASBLY-26

## Type of PR

- [ ] Bugfix
- [x] New feature
- [ ] Minor changes

## Changes introduced

while running any CB CLI command it prints out currently installed and latest version from registry of the CLI
It also warns the user that they have older version and provide a command to update the version

## Test and review

Run any/all commads and it should print something like this
`CLI Version current: 3.1.0 latest: 3.1.0`
or 
`⚠ CLI Version current: 3.0.0 latest: 3.1.0
⚠ You have an older version. Please update to new version by following this command: npm install -g crowdbotics`

depending of your state of environment

### CODE REVIEW

File: scripts/utils/environment.js [hunk 1]
```diff
@@ -179,4 +222,25 @@ export function validateEnvironmentDependencies(
       valid(environmentVersions.cookiecutter);
     }
   }
+
+  if (dependencies.includes(EnvironmentDependency.CLI)) {
```
Comment (taylor-cb) [inline scripts/utils/environment.js:226]: Let's tidy up the nested if statements if we can. Suggest returning early once conditions aren't met.

```typescript
 if (!environmentVersions.cli) {
  printInvalidMessage("cli is not available in your system");
  return;
}

if (!environmentVersions?.cli?.registry) {
  printInvalidMessage("cli is not available in your system");
  return;
}

const cliVersionMessage = `CLI Version current: ${environmentVersions?.cli?.local?.version} latest: ${environmentVersions?.cli?.registry?.latest}`;


```

File: utils.js [hunk 2]
```diff
@@ -23,3 +23,20 @@ export const section = (...args) => {
 };
 
 export const isUserEnvironment = !process?.env?.CI && !process?.env?.CIRCLE_JOB;
+
+export const compareVersions = (v1, v2) => {
```
Comment (taylor-cb) [inline utils.js:27]: Maybe using https://www.npmjs.com/package/semver might be safer here?

Comment (alaisgomes) [reply utils.js:27]: Semver package is quite small, it might simplify this part of the code too. I wouldn't mind trying it out for safety in the checks

Comment (shahraizali) [reply utils.js:27]: I agree, just didn't want to add a new dependency in my very first commit 😅

Comment (taylor-cb) [reply utils.js:27]: Ha! yes agreed. Small enough it's worth it I think.

**Review by alaisgomes — DISMISSED** [review]

**Review by alaisgomes — APPROVED** [review]

**Review by taylor-cb — APPROVED** [review]

**Pull request merged**